packages/circuit-ui/components/Checkbox/Checkbox.spec.tsx

```diff
@@ -31,37 +25,44 @@ const defaultProps = {
 };
 
 describe('Checkbox', () => {
+  // ResizeObserver Mock
+  global.ResizeObserver = jest.fn().mockImplementation(() => ({
+    observe: jest.fn(),
+    unobserve: jest.fn(),
+    disconnect: jest.fn(),
+  }));
```

Comment:
We can't rely on the `ResizeObserver` because it's not available in all [browsers that SumUp supports](https://github.com/sumup/legacy-browsers).

An and Robin found a workaround for the Popover (https://github.com/sumup-oss/circuit-ui/pull/1661), let's use the same here.

Reply:
I'm currently working on a solution related to Debouncer and Throttle. The pure debouncer solution isn't working when the user resizes the window because the constant flow of resize events resets the counter to update the tooltip position. The throttle solution on Circuit UI isn't helping because it makes sure that the resizing happens at most once during a period of time, but it's doesn't necessarily update in the final resize.